core/middlewares/with-auth.ts

```diff
@@ -1,7 +1,26 @@
+import { NextResponse } from 'next/server';
+
 import { auth } from '~/auth';
 
 import { type MiddlewareFactory } from './compose-middlewares';
 
-// The `auth` function doesn't have the correct type to support it as a MiddlewareFactory.
-// eslint-disable-next-line @typescript-eslint/consistent-type-assertions
-export const withAuth: MiddlewareFactory = auth as unknown as MiddlewareFactory;
+const AUTH_PATHS = ['/login/', '/register/', '/reset/', '/change-password/'];
+
+export const withAuth: MiddlewareFactory = (next) => {
+  return async (request, event) => {
+    // @ts-expect-error: The `auth` function doesn't have the correct type to support it as a MiddlewareFactory.
+    const authWithCallback = auth(async (req) => {
+      if (req.auth && AUTH_PATHS.includes(req.nextUrl.pathname)) {
+        return NextResponse.redirect(new URL('/account', req.nextUrl.origin));
+      }
+
+      // Continue the middleware chain
+      const response = await next(req, event);
+
+      return response;
```

Comment:
Since we are not doing anything with the response, could we `return next(req, event);` instead?